format/avro/schema/schema.go

```diff
@@ -0,0 +1,198 @@
+package schema
+
+import (
+	"errors"
+	"fmt"
+)
+
+const (
+	ARRAY   = "array"
+	BOOLEAN = "boolean"
+	BYTES   = "bytes"
+	DOUBLE  = "double"
+	ENUM    = "enum"
+	FIXED   = "fixed"
+	FLOAT   = "float"
+	INT     = "int"
+	LONG    = "long"
+	MAP     = "map"
+	NULL    = "null"
+	RECORD  = "record"
+	STRING  = "string"
+	UNION   = "union" // avro spec doesn't treat unions like this, but makes it easier for us
+)
+
+type SimplifiedSchema struct {
+	Type        string            `json:"type"`
+	Name        string            `json:"name"`
+	LogicalType string            `json:"logicalType,omitempty"`
+	Size        int               `json:"size,omitempty"`
+	Scale       int               `json:"scale,omitempty"`
+	Precision   int               `json:"precision,omitempty"`
+	Items       *SimplifiedSchema `json:"items,omitempty"`
+	Fields      []Field           `json:"fields,omitempty"`
+	Symbols     []string          `json:"symbols,omitempty"`
+	Values      *SimplifiedSchema `json:"values,omitempty"`
+	UnionTypes  []SimplifiedSchema
+	//Choosing not to handle Default as it adds a lot of complexity and this is used for showing the binary
+	//representation of the data, not fully parsing it. See https://github.com/linkedin/goavro/blob/master/record.go
+	//for how it could be handled.
+}
+
+type Field struct {
+	Name string
+	Type SimplifiedSchema
+}
+
+func SchemaFromJson(schema interface{}) (SimplifiedSchema, error) {
+	if schema == nil {
+		return SimplifiedSchema{}, errors.New("schema cannot be nil")
+	}
+	var s SimplifiedSchema
+	switch v := schema.(type) {
+	case []interface{}:
+		s.Type = UNION
+		for _, i := range v {
+			unionType, err := SchemaFromJson(i)
+			if err != nil {
+				return s, fmt.Errorf("error parsing union type: %s", err)
+			}
+			if unionType.Type == UNION {
+				return s, errors.New("sub-unions are not supported")
+			}
+			s.UnionTypes = append(s.UnionTypes, unionType)
+		}
+	case string:
+		s.Type = v
+	case map[string]interface{}:
```

Comment:
Heh I was looking at that before wasting so much time on writing this manually. But decided against it cause I didn't want to import anything that wasn't necessary into the project and wasn't sure how it handled recursion. Should have checked the mod file first 🙃 

Ill review it to see if it can handle recursion and update to use it if so.